site/_posts/2020-07-10-Contours-landscape-jul-2020.md

```diff
@@ -0,0 +1,112 @@
+ 
+With that said, here are some things the Contour team is going to do to try and improve.
+
+### Building greater configurability
+
+In order to recognize that a segment of our users really need more configurability, we are going to allow more configuration of a greater set of settings. In the past we’ve tried to set reasonable defaults, and add configurable settings as a last resort.
+
+We’re going to be building more configurability, and pushing that configurability towards Contour’s configuration file with the aim of exposing whatever complexity we need in there for Contour-level configuration (as opposed to individual-resource level configuration). We’ll be starting with timeouts, putting configuration of the timeouts requested in [contour#2225](https://github.com/projectcontour/contour/issues/2225) into the config file, with some thought put into both how to configure upper and/or lower bounds for those timeouts if necessary, and how to name the timeouts to make sense for people who only know Contour, not Envoy.
+
+Over time, we’ll try to limit the number of command line flags to Contour, with the aim of eventually paring Contour down to just the flags useful for fast local development (such as `--insecure` to not require TLS certs for xDS serving, or `--debug` to enable debug logging).
+
+The corollary to accepting greater configurability is that, whatever we do, we need to support. So, the requirement for these timeout settings will be better documentation of the configuration file, and documentation of where in a request chain the timeouts apply. More generally, I’ll be asking that designs that add open-ended configuration include documentation about why the default value was chosen, and what are reasonable values for at least some use cases.
+
+Other features similar to this include the addition of CORS support to HTTPProxy, and similar tweakables.
+
+### Exposing more Envoy configuration
+We’ve had a few requests for exposing more Envoy-specific configuration settings over Contour’s lifetime, and historically, we’ve been reluctant to break Contour’s abstraction in any substantial way, preferring to expose an abstracted version rather than the Envoy config directly. The exception to this is the bootstrap configuration - we currently supply `contour bootstrap` as a way to automatically generate this config, but it’s always been intended that Contour users could generate their own bootstrap configuration and add additional features if they wish. We’ve not done a great job of communicating this ability in the past.
```

Comment:
I think this falls into the same boat as Lua filters. We never intended users to pass their own bootstrap config since it's pretty easy to get stuck in a spot where everything gets broken. 

Also, many features that are enabled here still require dynamic configuration in Contour which means it should be just configured from there at the start.

Reply:
This is more to acknowledge that you *can* use this, not that it's a good idea. I'm not sure how to say that though.